src/main/java/com/robertx22/mine_and_slash/a_libraries/dmg_number_particle/particle/InteractionResultHandler.java

```diff
@@ -0,0 +1,118 @@
+package com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle;
+
+import com.google.common.collect.ImmutableMap;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.impl.DamageNullifiedParticle;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.impl.ElementDamageParticle;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.impl.HealParticle;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.style.Default;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.style.Row;
+import com.robertx22.mine_and_slash.uncommon.enumclasses.Elements;
+import com.robertx22.mine_and_slash.uncommon.utilityclasses.ClientOnly;
+import com.robertx22.mine_and_slash.uncommon.utilityclasses.NumberUtils;
+import com.robertx22.mine_and_slash.vanilla_mc.packets.interaction.IParticleSpawnMaterial;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.Minecraft;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.entity.Entity;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Random;
+import java.util.function.BiConsumer;
+
+public class InteractionResultHandler {
+
+
+
+    public enum ParticleSpawnType {
+        DAMAGE(new IParticleSpawnMaterial.DamageInformation(null, null, false)) ,
+        NULLIFIED_DAMAGE(IParticleSpawnMaterial.Type.DODGE) ,
+        HEAL(new IParticleSpawnMaterial.HealNumber(0.0f)) ;
+        public final IParticleSpawnMaterial target;
+
+        ParticleSpawnType(IParticleSpawnMaterial target) {
+            this.target = target;
+        }
+    }
+
+    public enum ClientReactionStrategy {
+        DEFAULT((info, entity) -> {
+            var mat = (IParticleSpawnMaterial.DamageInformation) info;
+            ImmutableMap<Elements, Float> dmgMap = mat.getDmgMap();
+
+            boolean crit = mat.isCrit();
+            for (Map.Entry<Elements, Float> entry : dmgMap.entrySet()) {
+                Float damage = entry.getValue();
+                if (damage
+                        .intValue() > 0) {
+
+                    double x = entity.getRandomX(0.5D);
+                    double y = entity.getEyeY();
+                    double z = entity.getRandomZ(0.5D);
+                    String damageString = NumberUtils.format(damage);
+                    Minecraft.getInstance().particleEngine.add(new ElementDamageParticle(Minecraft.getInstance().level, x, y, z, new Default(), entry.getKey().format.getColor(), crit ? damageString + "!" : damageString));
+                }
+            }
+        },
+                (type, entity) -> {
```

Comment:
Why isn't damageStrategy etc just a field in ParticleSpawnType  ?

Reply:
I don't get you, you mean to break down the ClientReactionStrategy?

Reply:
Yeah it should just tell ParticleSpawnType  the pos and other info to spawn the particle and ParticleSpawnType should be spawning the particle.

Reply:
The ClientReactionStrategy is not just a position picker. See the damageStrategy, the DEFAULT will generate multiple particles while the IN_A_ROW will generate only one. 

I also don't wanna merge these two enums, it will make the ParticleSpawnType very chunky and I can't imagine the way I can use config to easily switch the style after merging.